cmake/toolchain/clang/generic.cmake

```diff
@@ -0,0 +1,5 @@
+set(CLANG_ROOT $ENV{CLANG_ROOT_DIR})
+set_ifndef(CLANG_ROOT /usr)
+set(COMPILER clang)
+
+set(CMAKE_C_COMPILER          ${CLANG_ROOT}/bin/clang)
```

Comment:
setting `CMAKE_C_COMPILER` does not belong to a toolchain file, it belongs to the `cmake/compiler/${COMPILER}/{target,generic}.cmake` files.
To be clear, the toolchain files should set the `COMPILER` variable pointing to the cmake file responsible for finding and setting the actual compiler (`CMAKE_C_COMPILER`).
That two-stage rocket is a nice approach, and we will eventually follow the same for linker and binutils.

Reply:
clang is not a toolchain. Strictly speaking, clang is a parser.

It would clear up confusion if we named according to:

- `llvm` as the toolchain (clang, lld, libs, tools).
- `clang` as the compiler.

Reply:
I agree with all points, but due to problems with the clang toolchain[0]. I would prefer not to resolve these issues in this PR.

Note that these comments apply equally well before and after this PR, so they are out-of-scope of this PR.

[0] https://github.com/zephyrproject-rtos/zephyr/issues/11187#issuecomment-445245153